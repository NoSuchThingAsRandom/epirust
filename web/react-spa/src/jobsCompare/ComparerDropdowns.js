/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from "react";
import JobsDropdown from "./JobsDropdown";
import PropTypes from "prop-types";

export default function ComparerDropdowns({jobs, updateSelectedJobs}) {
  return (
    <div className='comparer-dropdowns container'>
      <div className='row justify-content-md-center'>
        <div className="form-group col-4">
          <JobsDropdown jobs={jobs} label='Job 1'/>
        </div>
        <div className="form-group col-4">
          <JobsDropdown jobs={jobs} label='Job 2'/>
        </div>
      </div>
    </div>
  );
}

ComparerDropdowns.propTypes = {
  jobs: PropTypes.array.isRequired,
  updateSelectedJobs: PropTypes.func
}