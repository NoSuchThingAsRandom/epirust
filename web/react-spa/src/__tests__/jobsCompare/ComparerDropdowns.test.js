/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import {fireEvent, render} from '@testing-library/react'
import ComparerDropdowns from "../../jobsCompare/ComparerDropdowns";

describe('Comparer Dropdowns', () => {
  it('should render inputs to select jobs to compare', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<ComparerDropdowns jobs={[{_id: 'id1'}, {_id: 'id2'}]}/>);

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  });

  it('should return selected jobs when compare button clicked', () => {
    const mockOnCompare = jest.fn()
    const {container} = render(<ComparerDropdowns jobs={[{_id: 'id1'}, {_id: 'id2'}]} onCompare={mockOnCompare}/>);

    const dropdowns = container.querySelectorAll('.form-control');
    fireEvent.change(dropdowns[0], {target: {value: 'id1'}})
    fireEvent.change(dropdowns[1], {target: {value: 'id2'}})
    fireEvent.click(container.querySelector('button'))

    expect(mockOnCompare).toHaveBeenCalledTimes(1)
    expect(mockOnCompare).toHaveBeenCalledWith({job1: 'id1', job2: 'id2'})
  });
});

