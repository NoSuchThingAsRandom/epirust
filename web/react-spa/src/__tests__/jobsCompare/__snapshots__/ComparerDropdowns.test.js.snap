// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparer Dropdowns should render inputs to select jobs to compare 1`] = `
<div
  className="comparer-dropdowns container"
>
  <form
    className="row justify-content-md-center"
    onSubmit={[Function]}
  >
    <div
      className="col-4"
    >
      <JobsDropdown
        jobs={
          Array [
            Object {
              "_id": "id1",
            },
            Object {
              "_id": "id2",
            },
          ]
        }
        label="Job 1"
      />
    </div>
    <div
      className="col-4"
    >
      <JobsDropdown
        jobs={
          Array [
            Object {
              "_id": "id1",
            },
            Object {
              "_id": "id2",
            },
          ]
        }
        label="Job 2"
      />
    </div>
    <div
      className="col-2 margin-top-auto"
    >
      <button
        className="btn btn-primary"
      >
        Compare
      </button>
    </div>
  </form>
</div>
`;
